sources/web/datalab/authentication.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/// <reference path="../../../externs/ts/node/node.d.ts" />
+/// <reference path="common.d.ts" />
+
+
+import logging = require('./logging');
+var googleapis = require('googleapis');
+
+/**
+ * The datastore service used to query Datastore to see if a user is allowed to access.
+ */
+var datastore : any;
+
+/**
+ * The authentication Url that it will redirect users to if needed.
+ */
+var authUrl: string;
+
+/**
+ * Checks whether a given user has access to this instance of Datalab.
+ */
+export function checkUserAccess(userId: string, cb: common.Callback<boolean>) {
+  datastore.datasets.lookup(
+    {
+      resource: {
+        keys: [{path: [{kind: 'DatalabUser', name: userId}]}]
+      }
+    },
+    function(e: Error, result: any) {
+      if (e) {
+        logging.getLogger().error(e, 'Failed to look up user.');
+      }
+      var found: boolean = result && result.found && (result.found.length > 0);
+      cb && cb(e, found);
+    }
+  );
+}
+
+/**
+ * Checks whether a given user has access to this instance of Datalab.
+ */
+export function getAuthenticationUrl(): string {
+  return authUrl;
+}
+
+export function init(settings: common.Settings, cb: common.Callback0) : void {
+  authUrl = "https://datalab.cloud.google.com?startinproject="
+            + settings.projectId + "&name=" + settings.instanceName;
+
+  var compute = new googleapis.auth.Compute();
```

Comment:
I am not sure how this works, but the token we get from here will expire in an hour, and the datastore wrapper created with it will no longer be usable after that.

Anyway, I think we do not need to create a datastore wrapper if we simply send the appropriate HTTP request. And effectively request the access token each time, so we get a valid token. The metadata service takes care of refreshing as needed.